src/components/SearchableSelect/SearchableSelect.jsx

```diff
@@ -0,0 +1,427 @@
+		const options = _.map(optionGroups, (optionGroupProps, optionGroupIndex) => {
+			const childOptions = _.map(_.get(optionGroupDataLookup, optionGroupIndex), ({ optionProps, optionIndex }) => (
+				this.renderOption(optionProps, optionIndex)
+			)).filter(childOption => childOption);
+
+			return (
+				childOptions.length > 0 &&
+				<DropMenu.OptionGroup key={'SearchableSelectOptionGroup' + optionGroupIndex} {...optionGroupProps}>
+					{optionGroupProps.children}
+					{childOptions}
+				</DropMenu.OptionGroup>
+			);
+		// then render all the ungrouped options at the end
+		}).concat(_.map(ungroupedOptionData, ({ optionProps, optionIndex }) => (
+			this.renderOption(optionProps, optionIndex)
+		))).filter(option => option);
+
+		return (
+			options.length > 0 ? options : <DropMenu.Option isDisabled><span className={cx('&-noresults')}>No results match "{searchText}"</span></DropMenu.Option>
+		);
+	},
+
+	render() {
+		const {
+			style,
+			className,
+			hasReset,
+			isDisabled,
+			isLoading,
+			isSelectionHighlighted,
+			maxMenuHeight,
+			searchText,
+			selectedIndex,
+			onSearch,
+			onSelect,
+			DropMenu: dropMenuProps,
+		} = this.props;
+
+		const {
+			direction,
+			flyOutStyle,
+			isExpanded,
+		} = dropMenuProps;
+
+		const {
+			flattenedOptionsData,
+		} = this.state;
+
+		const placeholderProps = _.first(_.map(findTypes(this.props, SearchableSelect.Placeholder), 'props'));
+		const placeholder = _.get(placeholderProps, 'children', 'Select');
+		const isItemSelected = _.isNumber(selectedIndex);
+
+		return (
+			<DropMenu
+				{...dropMenuProps}
+				className={cx('&', className)}
+				flyOutStyle={_.assign({}, flyOutStyle, !_.isNil(maxMenuHeight) ? { maxHeight: maxMenuHeight } : null)}
+				isDisabled={isDisabled}
+				isLoading={isLoading}
+				onSelect={onSelect}
+				selectedIndices={isItemSelected ? [selectedIndex] : []}
+				style={style}
+			>
+				<DropMenu.Control>
+					<div
+						tabIndex={0}
+						className={cx('&-Control', {
+							'&-Control-is-highlighted': (!isDisabled && isItemSelected && isSelectionHighlighted) || (isExpanded && isSelectionHighlighted),
+							'&-Control-is-selected': (!isDisabled && isItemSelected && isSelectionHighlighted) || (isExpanded && isSelectionHighlighted),
+							'&-Control-is-expanded': isExpanded,
+							'&-Control-is-disabled': isDisabled,
+						})
+					}>
+						<span
+							{...(!isItemSelected ? placeholderProps : null)}
+							className={cx(
+									'&-Control-content',
+									(!isItemSelected ? _.get(placeholderProps, 'className') : null)
+							)}
+						>
+							{isItemSelected ? flattenedOptionsData[selectedIndex].optionProps.children : placeholder}
+						</span>
+						<CaretIcon direction={isExpanded ? direction : 'down'} />
+					</div>
+					{
+						isExpanded &&
+						<div className={cx('&-Search-container')} onClick={this.handleClickSearchField}>
+							<SearchField
```

Comment:
Maybe I should expose a prop that allows for passing through `SearchField` props so that users of the component can have more control over the SearchField itself?